src/datatypes/minecraft.js

```diff
@@ -43,35 +43,36 @@ function writeUUID (value, buffer, offset) {
   return offset + 16
 }
 
-function readNbt (buffer, offset) {
-  return nbt.proto.read(buffer, offset, 'nbt')
+function readNbt (buffer, offset, { tagType } = { tagType: 'nbt' }) {
```

Comment:
Are we confident this is only useful for the network serialization?

Would there be any drawback to put this in prismarine-nbt ?

Reply:
This is already just directly calling prismarine-nbt so it should be possible to just import the prismarine-nbt types directly into the protocol. But the optionalNBT type may actually require a custom type as there is a special case with zero.

Reply:
Would require importing all the custom types also https://github.com/PrismarineJS/prismarine-nbt/blob/master/nbt.js#L11C1-L20C48

Reply:
It seems to me that would be much cleaner as all nbt stuff would then live in prismarine-nbt

Also important if other stuff than nmp use the same format, for example anvil